### PR TITLE
move from cra to next.js

### LINKED ISSUE #24: Move to Next.js
The current app is build with `cra` which outputs one bundle.
Since we plan to have more pages than the current one including some that will be for docs, it might be better to move to nextjs.
- [x] Migrate from cra to next.js
- [ ] #43
- [x] Setup directories and pages
- [ ] Ensure deployment works correctly.

### PR DESCRIPTION
**Pull request by geofmureithi**
- [x] Introduces Next.js
- [ ] Connect to nebula.eiger.co via Vercel

Closes #24

### CODE REVIEW

**Review by eloylp — COMMENTED** [review]
I have some questions :question: 

* What is `cra` and what is `next.js` ? - seriously, i have no idea :stuck_out_tongue: 
* How they will actually help to the project ? Why its not markdown enough ?

I tried to execute with `npm start`  but got:

```bash
❯ npm start

> nebula-demo@0.1.0 start
> next start

sh: 1: next: not found
```
I guess i need to install some binary first. Could you expand review instructions ?

**Comment by mariopil** [comment]
You need to run `npm install` first, and then `npm run build`

**Comment by geofmureithi** [comment]
You would need to build before starting.
For commands to work with please check package.json
See for explanations https://nextjs.org/docs/getting-started/installation#manual-installation

>What is cra and what is next.js ? - seriously, i have no idea 😛
-----> https://prismic.io/blog/create-react-app-cra-vs-nextjs

We are not using markdown yet but we will be able to use now with next.js.
We have a lot of docs and other static pages we need to build and next.js is handy in this.

File: app/next.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const withImages = require('next-images')
+module.exports = withImages({
+  webpack(config, options) {
+    return config
+  }
+})
```
Comment (mariopil) [inline app/next.config.js:6]: New line missing

File: app/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from "react"
+import Readme from "../src/Readme"
+// export default Index
+
+const Index = () => <Readme />
+
+export default Index
```
Comment (mariopil) [inline app/pages/index.tsx:7]: New line missing

File: app/pages/wizard.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from "react"
+import Wizard from "../src/Wizard"
+// export default Index
+
+const Page = () => <Wizard />
+
+export default Page
```
Comment (mariopil) [inline app/pages/wizard.tsx:7]: New line missing

**Review by mariopil — APPROVED** [review]

**Comment by eloylp** [comment]
> You need to run `npm install` first, and then `npm run build`

Noice thanks ! that solved part of the problem. Now i am getting:

```bash

❯ npm start  

> nebula-demo@0.1.0 start
> next start

- ready started server on [::]:3000, url: http://localhost:3000
[Error: ENOENT: no such file or directory, open '/home/user/projects/nebula/app/.next/BUILD_ID'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/user/projects/nebula/app/.next/BUILD_ID'
}

```

Any other command i am not aware ?

**Comment by mariopil** [comment]
Have you run `npm run build` before starting?

**Comment by eloylp** [comment]
> 



> Have you run `npm run build` before starting?

That did the trick ! :D thanks !

**Comment by geofmureithi** [comment]
Are we not going to create tutorials/docs of how to work with our contracts?

**Review by eloylp — APPROVED** [review]

**Comment by eloylp** [comment]
> Are we not going to create tutorials/docs of how to work with our contracts?

I dont know :)  , maybe still early. Maybe a plain README.md on each contract folder is enough. But i am open to try things in next.js.

**Pull request merged**